abuild-fetch.c

```diff
@@ -135,7 +142,7 @@ int fetch(char *url, const char *destdir)
 	if (access(outfile, F_OK) == 0)
 		goto fetch_done;
 
-	if (insecure) {
+	if (insecure || 0 == strncmp(http_scheme, scheme, sizeof(http_scheme))) {
```

Comment:
Why not something like:
````c
    /* enable insecure mode when http. This may be useful when it redirects to https */
    if (strstr(url, "http://") == url)
        insecure = true;
````

and leave all the rest as it was.